### ISSUE TITLE
Docker Daemon file descriptor leak

### ISSUE THREAD

**Issue by dmyerscough** [docker/docker#28495]
**Description**

We have been experiencing the docker daemon open the `/var/run/docker.sock` file over 1076718 and exhausting the total number of open file descriptors for the docker daemon.

**Steps to reproduce the issue:**
1. Running docker 1.12.1 for a few days in production

**Describe the results you received:**

I have provided a stacktrace here;-

https://gist.github.com/dmyerscough/2e6a407b1400511964857fc22c0a8e55

Kernel version is (4.4.27)

**Describe the results you expected:**

Docker should not be consuming so many open files (/var/run/docker.sock)

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:02:53 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:02:53 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 9
 Running: 0
 Paused: 0
 Stopped: 9
Images: 27
Server Version: 1.12.1
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null bridge host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options:
Kernel Version: 4.4.27
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 32
Total Memory: 251.8 GiB
Name: XXXXX-dca1
ID: P4FB:VAHF:FECY:IEAT:NM53:VWFU:4UFR:USWN:IWJ5:KRFU:74HB:V46A
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 10.0.0.0/8
 127.0.0.0/8```

**Additional environment details (AWS, VirtualBox, physical, etc.):** 

Physical Host

**Comment by thaJeztah** [comment]
ping @mlaventure PTAL!

**Comment by mlaventure** [comment]
Should be fixed by #27405
Duplicate of #25993

**Comment by thaJeztah** [comment]
Let's close this one then as a duplicate of https://github.com/docker/docker/issues/25993
